visualizer/utils/button.py

```diff
@@ -0,0 +1,220 @@
+import typing
+
+import pygame
+from visualizer.utils.text import Text
+from game.utils.vector import Vector
+from typing import Optional, Callable
+from typing import TypeAlias
+
+"""
+Class that creates an intractable button extending the Text class.
+
+Defaults same as defaults used in Text class.
+Must give an action the button can perform upon being clicked
+Can select three colors for both the text and button: default, mouse hover, and mouse clicked
+Can also select padding for bg button and amount the border_radius for the bg button
+
+Parameters:
+Text               :  All parameters from the Text class (screen, text, font_size, font_name, fg_color, position)
+action             :  Action performed when button is clicked
+fg_color           :  Used to store default text color              Default - #daa520
+fg_color_hover     :  Text color for hovering over button           Default - #fff000
+fg_color_clicked   :  Text color for clicking button                Default - #1ceb42
+bg_color           :  bg color default                              Default - #846311
+bg_color_hover     :  bg color for hovering over button             Default - #936e13
+bg_color_clicked   :  bg color for clicking button                  Default - #efb21b
+padding            :  Amount of padding given to bg rect            Default - 5
+border_radius      :  Level of smoothing to corners of button       Default - 5
+
+In future projects, defaults for button style should be changed according to style of game for ease of code
+"""
+
+# typing alias for color
+Color: TypeAlias = str | int | tuple[int, int, int, Optional[int]] | list[
+    int, int, int, Optional[int]] | pygame.Color
+
+
+class Button(Text):
+    def __init__(self, screen: pygame.Surface, text: str, action: Callable,
```

Comment:
I am going to go back on what I said; create a class to hold the 6 colors and extract the logic for the getters and setters there then use that class as a parameter for this, there are too many elements in this constructor.